tests/suites/test_suite_pk.function

```diff
@@ -611,7 +612,8 @@ void pk_sign_verify( int type, int parameter, int sign_ret, int verify_ret )
 {
     mbedtls_pk_context pk;
     size_t sig_len;
-    unsigned char hash[MBEDTLS_MD_MAX_SIZE];
+    unsigned char hash[32]; // Hard-coded for SHA256
```

Comment:
Note: this is very similar to the second commit of https://github.com/ARMmbed/mbedtls/pull/4561 - so perhaps we can close #4561 in favour of this one ? (But probably still backport the relevant parts of #4561, see my comments there.)

Reply:
I didn't look at #4561 while writing this, I'll take a look after #4665 is merged and I've rebased this.

Reply:
Since #4561 is a community PR and this PR (#4707) needs rebasing anyway, I think we could merge #4561 quickly and wait with rebasing #4707 until then. What do you think?

Reply:
#4561 is a behavior change where #4707 changes the behavior again, plus some testing improvements. I think it would be easier to put #4707 first and then make a version of #4561 with just the testing improvements, and perhaps some implementation improvements in `pk.c`, but no behavior change.